lib/descriptor-builder.js

```diff
@@ -172,15 +174,18 @@ DescriptorBuilder.prototype.setIdProperty = function(p, validate) {
   this.idProperty = p;
 };
 
-DescriptorBuilder.prototype.assertNotDefined = function(p, name) {
-  var propertyName = p.name,
+DescriptorBuilder.prototype.assertNotDefined = function(p) {
+  var ns = p.ns,
+      propertyName = p.name,
       definedProperty = this.propertiesByName[propertyName];
 
-  if (definedProperty) {
+  if (this.propertiesByName[ns.name] && this.propertiesByName[ns.localName]) {
```

Comment:
What should this check accomplish? Is there a situation where the property has been defined within the same `ns` and no local name assigned? If so, we should verify that would be actually fine.

Reply:
If `p` (property to be proofed) is defined in a different namespace than `this`, then `p.name` is the local name in THAT ns and `this.propertiesByName(p.name)` might already exist (mapping to a local property). In this case, the assertion needs to hold true. This allows the property from another ns to be added (`assertNotDefined()` is called by `addNamedProperty()`).

With this patch we allow the type to reference MULTIPLE (different) properties and therefore need to distinguish `ns.name` from `ns.localName`. They can be different now.